csharp/src/Google.Protobuf/Reflection/CustomOptions.cs

```diff
@@ -77,6 +80,7 @@ internal CustomOptions(IDictionary<int, IExtensionValue> values)
         /// <param name="field">The field to fetch the value for.</param>
         /// <param name="value">The output variable to populate.</param>
         /// <returns><c>true</c> if a suitable value for the field was found; <c>false</c> otherwise.</returns>
+        [RequiresUnreferencedCode(UnreferencedCodeMessage)]
```

Comment:
`CustomOptions` is obsolete so I just marked it as incompatible with trimming instead of finding a way to fix it.